src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -41,6 +41,8 @@ object RelationalSpec {
     new MatrixTypeSerializer
 
   def read(hc: HailContext, path: String): RelationalSpec = {
+    if (!hc.hadoopConf.isDir(path))
+      fatal(s"a MatrixTable or Table file is a directory; path '$path' is not")
```

Comment:
Tables and MatrixTables files are directories; path '$path' is not a directory.

Reply:
yes, this is clearer